overlord/devicestate/devicemgr.go

```diff
@@ -1877,53 +1877,17 @@ func (m *DeviceManager) Systems() ([]*System, error) {
 // including the model assertion, gadget details and encryption info
 // for the given system label.
 func (m *DeviceManager) SystemAndGadgetAndEncryptionInfo(wantedSystemLabel string) (*System, *gadget.Info, *EncryptionSupportInfo, error) {
-	if m.isClassicBoot {
-		return nil, nil, nil, fmt.Errorf("cannot get model and gadget information on a classic boot system")
-	}
-	// get current system as input for loadSeedAndSystem()
-	systemMode := m.SystemMode(SysAny)
-	currentSys, _ := currentSystemForMode(m.state, systemMode)
+	// TODO check that the system is not a classic boot one when the
+	// installer is not anymore.
```

Comment:
sorry, I originally read the check too quickly. We will have to support classic installer images for a long while, so not sure we can implemente the TODO. We'll need to see if there's anything we can double check or not